src/pkg/packager/dev.go

```diff
@@ -105,3 +107,30 @@ func (p *Packager) DevDeploy(ctx context.Context) error {
 	// cd back
 	return os.Chdir(cwd)
 }
+
+// Lint ensures a package is valid & follows suggested conventions
+func (p *Packager) Lint(ctx context.Context) (err error) {
+	config.CommonOptions.Confirm = true
+
+	if err := os.Chdir(p.cfg.CreateOpts.BaseDir); err != nil {
+		return fmt.Errorf("unable to access directory %q: %w", p.cfg.CreateOpts.BaseDir, err)
+	}
+
+	lintFindings, err := lint.Validate(ctx, p.cfg.CreateOpts)
+	if err != nil {
+		return fmt.Errorf("linting failed: %w", err)
+	}
+
+	if len(lintFindings) == 0 {
+		message.Successf("0 findings for %q", p.cfg.Pkg.Metadata.Name)
+		return nil
+	}
+
+	lint.PrintFindings(lintFindings, types.SevWarn, p.cfg.CreateOpts.BaseDir, p.cfg.Pkg.Metadata.Name)
```

Comment:
I would prefer that the printing is done by caller of this function.

Reply:
Yeah I don't really like print logic at the moment either. Do you think it would be worth creating a separate package outside of lint to store the `packageFinding` type and to do the printing to the user there? Most of what is in `src/pkg/package/lint/validate.go` feels like it should be it's own or a different package, but I'm not sure where that code should live.

Following the current codebase it would probably be in `message` beside things like `printCredentialTable`, though I'm not sure we want to keep following that pattern.

Reply:
I changed it to do the printing in that function